### PR TITLE
HPCC-21901 Ensure named DafilesrvProcess component settings used

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: fs/dafilesrv/dafilesrv.cpp [hunk 1]
```diff
@@ -386,13 +386,119 @@ int main(int argc,char **argv)
...
     IPropertyTree *keyPairInfo = nullptr;
     if (env)
     {
         StringBuffer dafilesrvPath("Software/DafilesrvProcess");
-        if (instanceName.length())
-            dafilesrvPath.appendf("[@name=\"%s\"]", instanceName.str());
+        if (componentName.length())
+            dafilesrvPath.appendf("[@name=\"%s\"]", componentName.str());
+        else
+            dafilesrvPath.append("[1]"); // in absence of name, use 1st
```
Comment (mckellyln) [inline fs/dafilesrv/dafilesrv.cpp]: Is this the primary/only functional change in this PR ? (the other changes being general refactoring)

**Comment by mckellyln** [comment]
@jakesmith one question to help me review

File: fs/dafilesrv/dafilesrv.cpp (hunk 1 above)
Comment (jakesmith) [reply fs/dafilesrv/dafilesrv.cpp]: @mckellyln - no, the main change is moving the command line argument parsing before the above happens.
Previously it was later and therefore componentName was not set (it is passed on the command line)

**Review by mckellyln — APPROVED** [review]

**Comment by mckellyln** [comment]
@jakesmith changes look ok.

File: fs/dafilesrv/dafilesrv.cpp [hunk 2]
```diff
@@ -487,119 +593,16 @@ int main(int argc,char **argv)
...
-            }
-        }
-        else
-            break;
-    }
-
     if (0 == logDir.length())
     {
-        getConfigurationDirectory(NULL,"log","dafilesrv",instanceName.str(),logDir);
+        getConfigurationDirectory(NULL,"log","dafilesrv",componentName.str(),logDir);
```
Comment (mckellyln) [inline fs/dafilesrv/dafilesrv.cpp]: indentation ?

**Comment by jakesmith** [comment]
@richardkchapman - please merge.

**Comment by richardkchapman** [comment]
@jakesmith Targetting 7.4 - is that correct?

**Comment by jakesmith** [comment]
@richardkchapman - yes that's fine.

**Pull request merged**